crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,759 @@
+use std::sync::RwLockWriteGuard;
+
+use crate::prelude::*;
+
+#[uniffi::export]
+impl SargonOS {
+    /// Performs initial transaction analysis for a given raw manifest, including:
+    /// 1. Extracting the transaction signers.
+    /// 2. Executing the transaction preview GW request.
+    /// 3. Running the execution summary with the manifest and receipt.
+    /// Maps relevant errors to ensure proper handling by the hosts.
+    pub async fn analyse_transaction_preview(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        self.perform_transaction_preview_analysis(
+            instructions,
+            blobs,
+            message,
+            is_wallet_transaction,
+            nonce,
+            notary_public_key,
+        )
+        .await
+    }
+}
+
+/// Part of the error message indicating the deposits are denied for the account.
```

Comment:
@sergiupuhalschi-rdx can you update this doc, to also document that it is an error message returned by Gateway.

```rust
/// This is part of an error message returned **by Gateway**, indicating the deposits are denied for the account.
/// We use it part of logic below, matching against this String - we really should upgrade this code to be more
/// structure - we MUST update this value if Gateway where to change this value.
```

or something like that. And same for `GW_ERR_NOT_ALL_COULD_BE_DEPOSITED`